File: C Programming (All Files)/C Programming (Lab Report 4)/Question_3.c
// Write a C Program To Define Multiple Macro's To Perform Arithmetic Operations.

#include <stdio.h>
#define ADD(x, y) ((x) + (y))
#define SUBTRACT(x, y) ((x) - (y))
#define MULTIPLY(x, y) ((x) * (y))
#define DIVIDE(x, y) ((float)(x) / (y))

int main()
{
    int num1;
    int num2;
    float result;

    printf("Please Enter The First Number : ");
    scanf("%d", &num1);

    printf("Please Enter The Second Number : ");
    scanf("%d", &num2);

    result = ADD(num1, num2);
    printf("The Addition Of The Given Numbers %d & %d Is : %.2f\n", num1, num2, result);

    result = SUBTRACT(num1, num2);
    printf("The Subtraction Of The Given Numbers %d & %d Is : %.2f\n", num1, num2, result);

    result = MULTIPLY(num1, num2);
    printf("The Multiplication Of The Given Numbers %d & %d Is : %.2f\n", num1, num2, result);

    if (num2 == 0)
    {
        printf("Divison By Zero Is Not Defined");
    }
    else
    {
        result = DIVIDE(num1, num2);
        printf("The Division Of The Given Number %d & %d Is : %.2f", num1, num2, result);
    }

    return 0;
}

File: Python Programming (All Files)/Experiment 10 (Data Analyze & Visualize)/Question_3.py
# Write a Python Program To Perform Matrix Multiplication Of Any 2 N x N Matrices.

import numpy as np

print("Enter The Size Of The Square Matrices (N x N) : ", end = "")
n = int(input())

print("Enter The Elements Of The First Matrix (Only {} Elements) : ".format(n*n), end = "")
num1 = [int(i) for i in input().split()]

print("Enter The Elements Of The Second Matrix (Only {} Elements) : ".format(n*n), end = "")
num2 = [int(i) for i in input().split()]

matrix1 = np.array(num1).reshape(n, n)
matrix2 = np.array(num2).reshape(n, n)
result = np.dot(matrix1, matrix2)

print("Result Of Matrix Multiplication : ")
print(result)